### PR TITLE
Clarify the rules on server-sent HEADERS

### LINKED ISSUE #878: Unpromised Push
As a corollary to https://github.com/quicwg/base-drafts/issues/4930, I cannot find anything in RFC 7540 that explicitly prohibits the server sending a response on a stream that hasn't been promised.  The closest is the state machine, where a push stream is supposed to go from idle to reserved to half-closed.  The statement that it can go from idle to open upon sending a HEADERS frame is clearly oriented at the client, but is endpoint-agnostic:  `Sending or receiving a HEADERS frame causes the stream to become "open".`

### PR DESCRIPTION
**Pull request by Lukasa**
Resolves #878.

### CODE REVIEW

**Comment by mnot** [comment]
Probably needs to be listed in "Changes from RFC7540".

**Comment by Lukasa** [comment]
Does it? I think this change is editorial: this is already _effectively_ the rule, we're just calling it out explicitly.

**Comment by mnot** [comment]
If by "effectively" you mean "it's what implementations do", we still need to document a change in the *rfc*.

**Comment by martinthomson** [comment]
I read that as "it's what the document originally said, just not very clearly".

**Comment by Lukasa** [comment]
Yup, that's my assertion, see the discussion in #878.

File: draft-ietf-httpbis-http2bis.xml [hunk 1]
```diff
@@ -727,6 +728,9 @@ HTTP Frame {
             <t>
                 Receiving any frame other than <xref target="HEADERS" format="none">HEADERS</xref> or <xref target="PRIORITY" format="none">PRIORITY</xref> on
                 a stream in this state MUST be treated as a <xref target="ConnectionErrorHandler">connection error</xref> of type
+                <xref target="PROTOCOL_ERROR" format="none">PROTOCOL_ERROR</xref>. If this stream is server-initiated as described in
+                <xref target="StreamIdentifiers"/> then receiving a <xref target="HEADERS" format="none">HEADERS</xref> frame MUST also
+                be treated as a <xref target="ConnectionErrorHandler">connection error</xref> of type
```
Comment (MikeBishop) [inline draft-ietf-httpbis-http2bis.xml:733]: Feels like it could use at least one comma in all that.

**Review by MikeBishop — APPROVED** [review]
Yes, I think this is correct.

File: draft-ietf-httpbis-http2bis.xml (hunk 1 above)
Comment (martinthomson) [reply draft-ietf-httpbis-http2bis.xml:733]:
```suggestion
                <xref target="PROTOCOL_ERROR" format="none">PROTOCOL_ERROR</xref>. If this stream is server-initiated, as described in
                <xref target="StreamIdentifiers"/>, then receiving a <xref target="HEADERS" format="none">HEADERS</xref> frame MUST also
                be treated as a <xref target="ConnectionErrorHandler">connection error</xref> of type
```
?

**Pull request merged**